tree/ntuple/test/ntuple_processor.cxx

```diff
@@ -143,79 +142,44 @@ TEST_F(RNTupleProcessorTest, Base)
 {
    auto proc = RNTupleProcessor::Create({fNTupleNames[0], fFileNames[0]});
 
-   int nEntries = 0;
-
-   for (const auto &entry : *proc) {
-      EXPECT_EQ(++nEntries, proc->GetNEntriesProcessed());
-      EXPECT_EQ(nEntries - 1, proc->GetCurrentEntryNumber());
+   auto x = proc->RegisterField<float>("x");
+   // Check that `RegisterField` also works with `void`.
+   auto y = proc->RegisterField<void>("y");
 
-      EXPECT_FLOAT_EQ(static_cast<float>(nEntries - 1), *entry.GetPtr<float>("x"));
-
-      std::vector<float> yExp{static_cast<float>(nEntries - 1), static_cast<float>((nEntries - 1) * 2)};
-      EXPECT_EQ(yExp, *entry.GetPtr<std::vector<float>>("y"));
+   try {
+      proc->RegisterField<float>("z");
+      FAIL() << "registering fields that do not exist should not be possible";
```

Comment:
This may not need be addressed in this PR, but in general this is possible. See e.g. https://github.com/root-project/root/blob/00565c678832d07c6c73cd2d732236d11c87b710/tree/dataframe/test/dataframe_incomplete_entries.cxx#L220

Users processing a vertical concatenation of N files may not know if field "x" is always present in all files, and in an unlucky scenario the field may be missing exactly from the first file.

Reply:
I see. In principle there's nothing holding us back from allowing users to request arbitrary fields and have their validity set to `false` until it is actually found on disk. I will look into this possibility further and address it in a separate PR.